### PR TITLE
Add initial recipe for RI (v3.6)

### PR DESCRIPTION
**Pull request by RJMW**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/ri/build.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+sed '2d' makefile
+
+CXX=$CXX
```
Comment (nsoranzo) [inline recipes/ri/build.sh:5]:
```suggestion
sed -i.bak '2d' makefile
```

File: recipes/ri/makefile.patch [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+-CFLAGS=-c -O3
+
+ SOURCES= ri3.cpp
+ OBJECTS=$(SOURCES:.cpp=.o)
+@@ -10,7 +8,7 @@
+ all:    $(SOURCES) $(EXECUTABLE)
+
+ $(EXECUTABLE): $(OBJECTS)
+-    $(CC) $(OBJECTS) -o $@
++    $(CC) $(CFLAGS) $(OBJECTS) -o $@
```
Comment (nsoranzo) [inline recipes/ri/makefile.patch:15]: This needs to be tabs.
```suggestion
-	$(CC) $(OBJECTS) -o $@
+	$(CC) $(CFLAGS) $(OBJECTS) -o $@
```

File: recipes/ri/makefile.patch [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+ .cpp.o:
+-    $(CC) $(CFLAGS) $< $(INCLUDES) -o $@  
++    $(CC) -c $(CFLAGS) $< $(INCLUDES) -o $@
```
Comment (nsoranzo) [inline recipes/ri/makefile.patch:19]:
```suggestion
-	$(CC) $(CFLAGS) $< $(INCLUDES) -o $@  
+	$(CC) -c $(CFLAGS) $< $(INCLUDES) -o $@
```

File: recipes/ri/makefile.patch (hunk 2 above)
Comment (RJMW) [reply recipes/ri/makefile.patch:15]: Yes - I was just applying the change. But you are quicker! 😄

File: recipes/ri/makefile.patch [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+-CFLAGS=-c -O3
+
+ SOURCES= ri3.cpp
+ OBJECTS=$(SOURCES:.cpp=.o)
+@@ -10,7 +8,7 @@
+ all:	$(SOURCES) $(EXECUTABLE)
+
+ $(EXECUTABLE): $(OBJECTS)
+-	$(CC) $(OBJECTS) -o $@
++	$(CC) $(CFLAGS) $(OBJECTS) -o $@
```
Comment (nsoranzo) [inline recipes/ri/makefile.patch:15]: Actually this should probably be:
```suggestion
+	$(CC) $(LDFLAGS) $(OBJECTS) -o $@
```

File: recipes/ri/makefile.patch [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+-CFLAGS=-c -O3
+
+ SOURCES= ri3.cpp
+ OBJECTS=$(SOURCES:.cpp=.o)
+@@ -10,7 +8,7 @@
+ all:	$(SOURCES) $(EXECUTABLE)
+
+ $(EXECUTABLE): $(OBJECTS)
+-	$(CC) $(OBJECTS) -o $@
++	$(CC) $(LDFLAGS) $(OBJECTS) -o $@
```
Comment (nsoranzo) [inline recipes/ri/makefile.patch:15]:
```suggestion
+	$(CXX) $(LDFLAGS) $(OBJECTS) -o $@
```

File: recipes/ri/makefile.patch [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+ .cpp.o:
+-	$(CC) $(CFLAGS) $< $(INCLUDES) -o $@  
++	$(CC) -c $(CFLAGS) $< $(INCLUDES) -o $@
```
Comment (nsoranzo) [inline recipes/ri/makefile.patch:19]:
```suggestion
+	$(CXX) -c $(CFLAGS) $< $(INCLUDES) -o $@
```

**Comment by nsoranzo** [comment]
Linux and macOS are green :tada:

File: recipes/ri/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+
+requirements:
+  build:
+    - cmake
+    - {{ compiler('cxx') }}
+    - {{ compiler('c') }}
+
+test:
+  commands:
+    - which ri36
```
Comment (scopatz) [inline recipes/ri/meta.yaml:24]: Is there a help command you can run?

File: recipes/ri/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+    - makefile.patch
+    
+build:
+  number: 0
+
+requirements:
+  build:
+    - cmake
+    - {{ compiler('cxx') }}
+    - {{ compiler('c') }}
```
Comment (nsoranzo) [inline recipes/ri/meta.yaml:20]:
```suggestion
```

File: recipes/ri/meta.yaml (hunk 7 above)
Comment (RJMW) [reply recipes/ri/meta.yaml:24]: No - the command `ri36` itself prints a list with options, but for some reason it fails the test.

**Comment by RJMW** [comment]
@scopatz many thanks for your help.

I would like to build ri for windows as well. Please, can you give me some guidance on what to include in bld.bat as I assume `make` is not compatible with windows? Do you have an example recipe that I can look at?

**Comment by scopatz** [comment]
make actually is on windows! But you should probably look at the project's instructions for building on windows and put those in bld.bat

**Comment by nsoranzo** [comment]
2020-06-12T22:09:32.4166930Z ./include/timer.h(66): fatal error C1083: Cannot open include file: 'sys/time.h': No such file or directory
2020-06-12T22:09:32.4168692Z makefile:13: recipe for target 'ri3.o' failed

File: recipes/ri/build.sh [hunk 9]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+sed -i.bak '2d' makefile
```
Comment (nsoranzo) [inline recipes/ri/build.sh:3]: I think this line can be removed (also from bld.bat), given we are using makefile.patch now.

**Pull request closed without merging**